src/js/sjcl-custom/sjcl-validecc.js

```diff
@@ -1,21 +1,130 @@
-sjcl.ecc.ecdsa.secretKey.prototype.sign = function(hash, paranoia, k_for_testing) {
+'use strict';
+
+/* eslint-disable no-undef */
+/* eslint-disable block-scoped-var */
+/* eslint-disable new-cap */
+/* eslint-disable valid-jsdoc */
+
+var BITS_ONE_BYTE = sjcl.codec.bytes.toBits([0x01]);
+var BITS_ZERO_BYTE = sjcl.codec.bytes.toBits([0x00]);
+var BITS_32_ZERO_BYTES = sjcl.codec.bytes.toBits(arrayFill(32, 0));
+var BITS_32_ONE_BYTES = sjcl.codec.bytes.toBits(arrayFill(32, 1));
+
+function arrayFill(n, fill) {
+  var arr = new Array(n);
+  for (var i = 0; i < n; i++) {
+    arr[i] = fill;
+  }
+  return arr;
+}
+
+function isZero(bn) {
+  return bn.limbs.length === 1 && bn.limbs[0] === 0;
+}
+
+function isPointInfinity(affine) {
+  var jac = affine.toJac();
+  return isZero(jac.z) && !isZero(jac.y);
+}
+
+function sha256Mac(k) {
+  var hmac = new sjcl.misc.hmac(k, sjcl.hash.sha256);
+  Array.prototype.slice.call(arguments, 1).forEach(hmac.update.bind(hmac));
+  return hmac.digest();
+}
+
+// https://tools.ietf.org/html/rfc6979#section-3.2
```

Comment:
This was basically ported to sjcl from bitcoinjs: https://github.com/bitcoinjs/bitcoinjs-lib/blob/master/src/ecdsa.js#L77. In the process of adding 6979 to ripple-lib-java, I checked the outputs of this against bouncy castle 1.52, with the k256 curve ( which happens to fully implement the spec, and all the test vectors ) and they produced the same output.